src/components/figures/chart/chartJs/chartjs_funnel_chart.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  BarControllerChartOptions,
+  BarControllerDatasetOptions,
+  BarElement,
+  CartesianParsedData,
+  CartesianScaleTypeRegistry,
+  TooltipPositionerFunction,
+} from "chart.js";
+
+export class FunnelChartController extends window.Chart?.BarController {
+  static id = "funnel";
+  static defaults = { ...window.Chart?.BarController.defaults, dataElementType: "funnel" };
+
+  updateElements(rects, start, count, mode) {
+    super.updateElements(rects, start, count, mode);
+    for (let i = start; i < start + count; i++) {
+      const rect = rects[i];
+      this.updateElement(rect, i, { nextElementWidth: rects[i + 1]?.width || 0 }, mode);
```

Comment:
could you add a small comment to explain what this does? for people who never went into chartjs internals